phys2denoise/metrics/retroicor.py

```diff
@@ -0,0 +1,87 @@
+"""implementation of retroicor"""
+
+import argparse as ap
+import numpy as np
+import scipy as sp
+import json
+import string
+import random
+import matplotlib as mpl; mpl.use('TkAgg')
+import matplotlib.pyplot as plt
+from scipy.signal import argrelmax
+
+"""
+This function computes RETROICOR regressors (Glover et al. 2000) 
+"""
+
+
+def compute_card_phase(card_peaks_timings,slice_timings,nscans,TR):
+    """
+
+    This function creates cardiac phase from cardiac peaks.
+    Assumes that timing of cardiac events are given in same units
+    as slice timings, for example seconds.
+    
+    """
+
+    nscans = np.shape(slice_timings)
+    phase_card = np.zeros(nscans)
+    for ii in range(nscans):
+        # find previous cardiac peaks
+        previous_card_peaks = np.asarray(np.nonzero(card_peaks_timings < slice_timings[ii]))
+        if np.size(previous_card_peaks) == 0:
+            t1 = 0
+        else:
+            last_peak = previous_card_peaks[0][-1]
+            t1 = card_peaks_timings[last_peak]
+        
+        # find posterior cardiac peaks
+        next_card_peaks = np.asarray(np.nonzero(card_peaks_timings > slice_timings[ii]))
+        if np.size(next_card_peaks) == 0:
+            t2 = nscans * TR
+        else:
+            next_peak = next_card_peaks[0][0]
+            t2 = card_peaks_timings[next_peak]
+        
+        # compute cardiac phase
+        phase_card[ii] = (2*np.math.pi*(slice_timings[ii] - t1))/(t2-t1)
+
+    return phase_card
+
+def compute_resp_phase(resp,sampling_time):
```

Comment:
NeuroKit2 has a function for calculating respiratory phase: https://github.com/neuropsychology/NeuroKit/blob/a04760bd83c544253ca919d214a46eb70cc06c39/neurokit2/rsp/rsp_phase.py#L9. Would that work here?